src/com/twu/biblioteca/Book.java

```diff
@@ -0,0 +1,35 @@
+package com.twu.biblioteca;
+
+public class Book {
+    private final String title;
+    private final String author;
+    private final int yearPublished;
+    private boolean checkedOut;
+
+    public Book(String title, String author, int yearPublished, boolean checkedOut) {
+        this.title = title;
+        this.author = author;
+        this.yearPublished = yearPublished;
+        this.checkedOut = false;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public String getAuthor() {
+     return author;
+   }
+
+   public int getYearPublished() {
+      return yearPublished;
+   }
+
+   public boolean getCheckedOut() {
+        return checkedOut;
+   }
+
+   public void setCheckedOut() {
```

Comment:
Is there another name we could use that also describes what is happening here?  setXXX is the convention for a setter method for attribute XXX, but it is  also typical to pass the value that XXX is being set to.

Reply:
public void setCheckedOut(Boolean checkedOut){
  this.checkedOut = true;
}

or should it be:

private boolean available; (or make it notAvailable to match my current intention)

public boolean isAvailable(){
return this.available;
}

public void setAvailable(boolean available){
  this.available = available;
}

but would the default state be false?

Reply:
I would land on just one term (i.e., either "checked out" or "available") and ideally the one that corresponds to the terminology used in the requirements/stories.  Unfortunately, the stories mention both terms, but it seems like "availability" is the status of the book and "checking out" is the verb applied to change that status.

Depending on how far down this thought experiment you want to go, you could consider whether keeping them independent makes sense.  In this case, the attribute would be "available" and it would have it's own set of getter/setter __and__ you would _also_ provide a `Book#checkOut()` method that checks whether the book is available.  _If there are any future requirements_ that talk of other events which result in a book being unavailable (e.g., being repaired), then this might make sense.

There's no right/wrong answers in any of these options, just degrees of simplicity of implementation, simplicity of readability and natural fit with the domain language.  Rarely can you optimise perfectly across all of these aspects simultaneously.